upgrade: Stop waiting on failed step

skazi0 opened the pull request:
Status API returns success HTTP code when current step has failed.
The actual status inside response needs to be checked to handle
such cases properly.
If that happens `onError` callback is triggered with the whole
response received from the API. Status code and response content
can be used to distinguish between different error cases.

BONUS: the tests were cleaned up a bit

skazi0 committed:
upgrade: Stop waiting on failed step

Status API returns success HTTP code when current step has failed.
The actual status inside response needs to be checked to handle
such cases properly.
If that happens `onError` callback is triggered with the whole
response received from the API. Status code and response content
can be used to distinguish between different error cases.

Itxaka reviewed (approved):
Looks good!

skazi0 commented:
![](https://github.trello.services/images/mini-trello-icon.png) [(3) UI: Generic error handling (admin backup)](https://trello.com/c/KwAOzuC2/205-3-ui-generic-error-handling-admin-backup)

matelakat reviewed (approved):
Looks good to me. I think I would have preferred the incomplete -> partial refactor step to be in a different commit.

skazi0 committed:
upgrade: Refactor upgradeStatusFactory tests

partialUpgradeResponse and incompleteUpgradeResponse were redundant.
Some tests were reordered to better match descriptions.

skazi0 commented:
@matelakat good point, I split the changes into two commits.
